lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
yes!  with one exception. The ruby setnx method returns boolean, not integers. So the XOR operator probably won't work as intended.